### PR TITLE
Parsing vb forms

### PR DESCRIPTION
**Pull request by MDoerner**
This PR changes the parser and lexer to enable us to parse VB6 forms.

If there are more constructs that have to be added, we can add them once we find an example we cannot parse.

### CODE REVIEW

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 1]
```diff
@@ -30,6 +30,7 @@ module :
     moduleAttributes
     moduleHeader?
     moduleAttributes
+	moduleConfigReferences?
```
Comment (bclothier) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: tab/spaces?

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 2]
```diff
@@ -687,6 +698,7 @@ keyword :
      | ARRAY
      | ATTRIBUTE
      | BEGIN
+	 | BEGINPROPERTY
```
Comment (bclothier) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: tab/spaces?

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 3]
```diff
@@ -713,6 +725,7 @@ keyword :
      | DOEVENTS
      | DOUBLE
      | END
+	 | ENDPROPERTY
```
Comment (bclothier) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: tab/spaces?

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 4]
```diff
@@ -42,14 +43,24 @@ module :
 
 moduleHeader : VERSION whiteSpace numberLiteral whiteSpace? CLASS? endOfStatement;
 
+moduleConfigReferences: moduleConfigReferenceElement+;
+
+moduleConfigReferenceElement: unrestrictedIdentifier whiteSpace? EQ whiteSpace? STRINGLITERAL whiteSpace? SEMICOLON whiteSpace? STRINGLITERAL endOfStatement;
```
Comment (bclothier) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: I think I'd rather have literal `OBJECT` than `unrestrictedIdentifier` for the LHS part. This has potential to hide bugs or something worse if we allow for a loose interpretation.

**Comment by MDoerner** [comment]
We will have to kind of parse it anyway. Simply removing the initial part completely is not an option as anything not in the token stream is lost on rewrite. Accordingly, we would destroy the form with every refactoring.

Moreover, to deal with it like with conditional compilation directives also requires to parse it in the precompilation parser in order to know which to know which tokens to hide.

In addition, we already had rules in the grammar for module configurations, which had not been sufisticated enough to deal with VBA forms. However, they might be necessary for some exported VBA code. I would rather not like to risk breaking existing functionality here. 

All in all, simply parsing the forms seems to me to be the easiest and most robust solution to the problem.

File: Rubberduck.Parsing/Grammar/VBAParser.g4 (hunk 2 above)
Comment (MDoerner) [reply Rubberduck.Parsing/Grammar/VBAParser.g4]: Oh, I hate that the grammar uses spaces. Will correct it.

**Comment by WaynePhillipsEA** [comment]
With the absence of the specification, I respectfully disagree.  Removing the vb forms text and restoring it before any rewrite seems far simpler than trying to parse something that we don't actually want to have to parse, and don't have the spec for.

File: Rubberduck.Parsing/Grammar/VBAParser.g4 (hunk 3 above)
Comment (Vogel612) [reply Rubberduck.Parsing/Grammar/VBAParser.g4]: While we're on that: @retailcoder also introduced a tabs vs spaces mismatch in the PR to exclude certain identifiers from linelabels

Comment (MDoerner) [reply Rubberduck.Parsing/Grammar/VBAParser.g4]: I will go through the entire file and replace all the tabs with the usual number of spaces; there are a lot of mismatches.

**Comment by bclothier** [comment]
@WaynePhillipsEA  

It may help to consider that parsing it does not mean actually consuming it. ATM, we don't really do much with that section. So even if there were some inaccuracy or mismatch with the unknown specs, parser would merely have wrong tokens assigned. Because we don't consume the result, it hardly matter and we don't need to manually work around the rewriting API for that section of text. 

For our use case, it's enough to handle the top-level and block statements so we know where the section starts/ends, then for the rest of process, we basically just ignore that block and not analyze those tokens within the RD codebase.

**Pull request merged**